src/genie/libs/parser/iosxr/show_isis.py

```diff
@@ -0,0 +1,172 @@
+"""
+show_isis.py
+
+IOSXR parsers for the following show commands:
+    * show isis adjacency
+    * show isis neighbors
+
+"""
+
+# Python
+import re
+from netaddr import IPAddress, IPNetwork
+
+# Metaparser
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Schema, Any, Or, Optional
+from genie.libs.parser.utils.common import Common
+
+
+#==================================
+# Schema for 'show isis adjacency'
+#==================================
+class ShowIsisAdjacencySchema(MetaParser):
+    """Schema for show run isis adjacency"""
+        
+    schema = {
+        'isis': {
+            Optional(Any()): {
+                Optional('adjacency'): {
+                    Optional(Any()): {
+                        Optional('total_adjacency_count'): int,
+                        Optional(Any()): {
+                            Optional('interface'): str,
+                            Optional('snpa'): str,
+                            Optional('state'): str,
+                            Optional('hold'): str,
+                            Optional('changed'): str,
+                            Optional('nsf'): str,
+                            Optional('ipv4_bfd'): str,
+                            Optional('ipv6_bfd'): str,
+                        },
+                    },
+                },
+            },
+        }
+    }
+
+class ShowIsisAdjacency(ShowIsisAdjacencySchema):
+    """Parser for show isis adjacency"""
+    
+    cli_command = 'show isis adjacency'
+    
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+        
+        isis_adjacency_dict = {}
+        isis_adjacency_dict['isis'] = {}
+        for line in out.splitlines():
+            line = line.rstrip()
```

Comment:
if you use line = line.strip() you won't need the '\s' at the beginning of every regexp.